packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -1040,7 +1040,11 @@ class _AdbLogReader extends DeviceLogReader {
     ];
     // logcat -T is not supported on Android releases before Lollipop.
     const int kLollipopVersionCode = 21;
-    final int apiVersion = int.tryParse(await device._apiVersion);
+    final int apiVersion = (String v) {
+      // If the API version string isn't found, conservatively assume that the
+      // version is less recent than the one we're looking for.
+      return v == null ? kLollipopVersionCode - 1 : int.tryParse(v);
```

Comment:
Do we know anything about the situations where apiVersion is showing up as null?
Is something going wrong when the tool tries to run `adb getprop`, or is the `ro.build.version.sdk` property actually missing?

Reply:
It's most likely that this property is just missing. If `adb getprop` had failed more severely, then the tool would not have gotten this far. In particular, it wouldn't have been able to figure out the `device.targetPlatform`:

https://github.com/flutter/flutter/blob/068fa84c5bb28382853a0506395158afa30f3cc6/packages/flutter_tools/lib/src/resident_runner.dart#L413

which boils down to a `getprop` call on Android:

https://github.com/flutter/flutter/blob/068fa84c5bb28382853a0506395158afa30f3cc6/packages/flutter_tools/lib/src/android/android_device.dart#L192